src/production/phases/FinalPhase.ts

```diff
@@ -1,20 +1,43 @@
+import { BundleRouter } from '../../bundle/bundleRouter';
+import { BUNDLE_RUNTIME_NAMES } from '../../bundleRuntime/bundleRuntimeCore';
 import { IProductionContext } from '../ProductionContext';
 
 /**
  * @todo:
- * - Create a new phase
- * - Loop modules and call transpile()
- * - fix require statements
- * - call generate()
- * - pipe that shit BundleRouter
  * - fix bundle router to understand splitEntry
  */
+export async function FinalPhase(productionContext: IProductionContext) {
+  const { ctx, entries, modules, splitEntries } = productionContext;
 
-export function FinalPhase(productionContext: IProductionContext) {
-  console.log(productionContext.splitEntries);
-  for (const module of productionContext.modules) {
+  await ctx.ict.resolve();
+
+  for (const module of modules) {
     if (module.isExecutable) {
-      module.transpile();
+      const transformerResult = module.transpile();
+
+      for (const item of transformerResult.requireStatementCollection) {
+        item.statement.callee.name = BUNDLE_RUNTIME_NAMES.ARG_REQUIRE_FUNCTION;
+        item.statement.arguments[0].value = module.moduleSourceRefs[item.statement.arguments[0].value].id;

```

Comment:
so many things can go wrong here ;-)

Reply:
i know.. will fix this (added a comment)